source/CompactRegular.lagda

```diff
@@ -0,0 +1,402 @@
+                                          → (U : ⟨ F ⟩)
+                                          → (U ⋜[ F ] U ⇒ is-clopen F U) holds
+well-inside-oneself-implies-clopen F U =
+ ∥∥-rec (holds-is-prop (is-clopen F U)) id
+
+clopenness-equivalent-to-well-inside-oneself : (F : frame 𝓤 𝓥 𝓦)
+                                             → (U : ⟨ F ⟩)
+                                             → (U ⋜[ F ] U) holds
+                                             ≃ is-clopen F U holds
+clopenness-equivalent-to-well-inside-oneself F U =
+   well-inside-oneself-implies-clopen F U
+ , logically-equivalent-props-give-is-equiv
+    (holds-is-prop (U ⋜[ F ] U))
+    (holds-is-prop (is-clopen F U))
+    (well-inside-oneself-implies-clopen F U)
+    (clopen-implies-well-inside-oneself F U)
+
+\end{code}
+
+\section{Definition of regularity}
+
+\begin{code}
+
+↓↓[_] : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → Fam (𝓤 ⊔ 𝓥) ⟨ F ⟩
+↓↓[ F ] U = (Σ V ꞉ ⟨ F ⟩ , (V ≤[ poset-of F ] U) holds) , pr₁
+
+\end{code}
+
+\begin{code}
+
+isRegular : frame 𝓤 𝓥 𝓦 → Ω (𝓤 ⊔ 𝓥)
+isRegular F = Ɐ U ∶ ⟨ F ⟩ , U is-lub-of (↓↓[ F ] U)
+ where
+  open Joins (λ U V → U ≤[ poset-of F ] V)
+
+\end{code}
+
+\section{Some properties}
+
+\begin{code}
+
+∨-is-scott-continuous : (F : frame 𝓤 𝓥 𝓦)
+                      → (U : ⟨ F ⟩)
+                      → is-scott-continuous F F (λ - → U ∨[ F ] -) holds
+∨-is-scott-continuous F U S dir = β , γ
+ where
+  open PosetNotation  (poset-of F) using (_≤_)
+  open PosetReasoning (poset-of F)
+  open Joins _≤_
+
+  β : ((U ∨[ F ] (⋁[ F ] S)) is-an-upper-bound-of ⁅ U ∨[ F ] Sᵢ ∣ Sᵢ ε S ⁆) holds
+  β i = ∨[ F ]-right-mono (⋁[ F ]-upper S i)
+
+  γ : (Ɐ (u′ , _) ∶ upper-bound ⁅ U ∨[ F ] Sᵢ ∣ Sᵢ ε S ⁆ ,
+        ((U ∨[ F ] (⋁[ F ] S)) ≤ u′)) holds
+  γ (u′ , p) = ∨[ F ]-least γ₁ γ₂
+   where
+    δ₁ : index S → (U ≤ u′) holds
+    δ₁ i = U                  ≤⟨ ∨[ F ]-upper₁ U (S [ i ]) ⟩
+           U ∨[ F ] (S [ i ]) ≤⟨ p i                       ⟩
+           u′                 ■
+
+    γ₁ : (U ≤[ poset-of F ] u′) holds
+    γ₁ = ∥∥-rec (holds-is-prop (U ≤[ poset-of F ] u′)) δ₁ (pr₁ dir)
+
+    γ₂ : ((⋁[ F ] S) ≤[ poset-of F ] u′) holds
+    γ₂ = ⋁[ F ]-least S (u′ , δ₂)
+     where
+      δ₂ : (u′ is-an-upper-bound-of S) holds
+      δ₂ i = S [ i ]                         ≤⟨ ∨[ F ]-upper₂ U (S [ i ]) ⟩
+             U ∨[ F ] (S [ i ])              ≤⟨ p i                       ⟩
+             u′                              ■
+
+∨-is-scott-continuous-eq : (F : frame 𝓤 𝓥 𝓦)
+                         → (U : ⟨ F ⟩)
+                         → (S : Fam 𝓦 ⟨ F ⟩)
+                         → (is-directed (poset-of F) S) holds
+                         → U ∨[ F ] (⋁[ F ] S) ≡ ⋁[ F ] ⁅ U ∨[ F ] Sᵢ ∣ Sᵢ ε S ⁆
+∨-is-scott-continuous-eq F U S dir =
+ ⋁[ F ]-unique ⁅ U ∨[ F ] Sᵢ ∣ Sᵢ ε S ⁆ (U ∨[ F ] (⋁[ F ] S)) (γ , δ)
+  where
+   γ = pr₁ ((∨-is-scott-continuous F U) S dir)
+   δ = pr₂ ((∨-is-scott-continuous F U) S dir)
+
+⋜₀-implies-≪-in-compact-frames : (F : frame 𝓤 𝓥 𝓦)
```

Comment:
Should not use 0 here.

Reply:
I am leaving it as it is because there is another function for the truncated version as well.